app/src/main/java/me/magnum/melonds/ui/common/LayoutComponentView.kt

```diff
@@ -2,30 +2,34 @@ package me.magnum.melonds.ui.common
 
 import android.view.View
 import android.widget.FrameLayout
+import androidx.core.view.updateLayoutParams
 import me.magnum.melonds.domain.model.LayoutComponent
 import me.magnum.melonds.domain.model.Point
 import me.magnum.melonds.domain.model.Rect
 
 class LayoutComponentView(val view: View, val aspectRatio: Float, val component: LayoutComponent) {
     fun setPosition(position: Point) {
-        val newParams = FrameLayout.LayoutParams(view.width, view.height).apply {
-            leftMargin = position.x
-            topMargin = position.y
+        view.updateLayoutParams {
+            FrameLayout.LayoutParams(view.width, view.height).apply {
```

Comment:
After a second revision, I found that these `updateLayoutParams` calls are being incorrectly used. This makes it impossible to drag or resize components in the layout editor. The proper usage of this method seems to be to modify the layout params received as `this`. For example, here, the correct usage would be the following:

```kt
view.updateLayoutParams<FrameLayout.LayoutParams> {
    this.leftMargin = position.x
    this.topMargin = position.y
}
```

The same logic applies to the `setSize()` and `setPositionAndSize()` methods. Notice that the generic parameter is only required if you need to access fields that are not present in the `ViewGroup.LayoutParams`. For example, you need to specify the concrete type here to access both `leftMargin` and `topMargin`.

Reply:
Good catch!